From 6df7bfad95ec562c0eef8dfeaf35a2854e9eef3e Mon Sep 17 00:00:00 2001
From: Mark Pictor <mpictor@gmail.com>
Date: Sun, 15 May 2011 19:35:31 -0400
Subject: Prepend OCE_ to names of HAVE_LIMITS and HAVE_LIMITS_H

Add test and macro for climits/HAVE_CLIMITS because Standard_values.h
does not like the limits header
---
 CMakeLists.txt             |    1 +
 inc/Standard_values.h      |    6 ++++--
 oce_build_config.h.cmake   |    1 +
 oce_install_config.h.cmake |    1 +
 4 files changed, 7 insertions(+), 2 deletions(-)

From 35e04906e2a2175eff8b499ee0bebf611d85222d Mon Sep 17 00:00:00 2001
From: Mark Pictor <mpictor@gmail.com>
Date: Sun, 15 May 2011 15:33:08 -0400
Subject: [PATCH 10/10] Prepend OCE_ to names of HAVE_IOSTREAM and
 HAVE_IOSTREAM_H macros in all files

---
 CMakeLists.txt                                     |    4 ++--
 inc/Standard_Stream.hxx                            |    6 +++---
 oce_build_config.h.cmake                           |    4 ++--
 oce_install_config.h.cmake                         |    4 ++--
 .../BinLDrivers_DocumentRetrievalDriver.cxx        |    2 +-
 5 files changed, 10 insertions(+), 10 deletions(-)

From db9d847c81e16bfe5f3dd0235aecb5d904bbc78d Mon Sep 17 00:00:00 2001
From: Mark Pictor <mpictor@gmail.com>
Date: Sun, 15 May 2011 19:39:00 -0400
Subject: [PATCH 07/10] Prepend OCE_ to names of HAVE_LIMITS and HAVE_LIMITS_H
 macros in all files

---
 CMakeLists.txt                        |    4 ++--
 inc/Standard_values.h                 |    6 +++---
 oce_build_config.h.cmake              |    4 ++--
 oce_install_config.h.cmake            |    4 ++--
 src/Draw/Draw_BasicCommands.cxx       |    4 ++--
 src/Graphic3d/Graphic3d_Structure.pxx |    4 ++--
 src/OSD/OSD_Chronometer.cxx           |    4 ++--
 7 files changed, 15 insertions(+), 15 deletions(-)

From 67f074f27349cee56adb32e6607179deba5d90db Mon Sep 17 00:00:00 2001
From: Mark Pictor <mpictor@gmail.com>
Date: Sun, 15 May 2011 15:24:08 -0400
Subject: [PATCH 08/10] Prepend OCE_ to names of HAVE_IOMANIP and
 HAVE_IOMANIP_H macros in all files

---
 CMakeLists.txt             |    4 ++--
 inc/Standard_Stream.hxx    |    4 ++--
 oce_build_config.h.cmake   |    4 ++--
 oce_install_config.h.cmake |    4 ++--
 4 files changed, 8 insertions(+), 8 deletions(-)

From f1505ad01235ed24837f2836ca9fe9b2ef929d17 Mon Sep 17 00:00:00 2001
From: Mark Pictor <mpictor@gmail.com>
Date: Sun, 15 May 2011 15:18:43 -0400
Subject: [PATCH 09/10] Prepend OCE_ to names of HAVE_FSTREAM and
 HAVE_FSTREAM_H macros in all files

---
 CMakeLists.txt             |    4 ++--
 inc/Standard_Stream.hxx    |    4 ++--
 oce_build_config.h.cmake   |    4 ++--
 oce_install_config.h.cmake |    4 ++--
 4 files changed, 8 insertions(+), 8 deletions(-)

Index: oce/inc/Standard_Stream.hxx
===================================================================
--- oce.orig/inc/Standard_Stream.hxx
+++ oce/inc/Standard_Stream.hxx
@@ -24,24 +24,24 @@
 // Unix variant
 #ifndef WNT
 
-#ifdef HAVE_IOSTREAM
+#ifdef OCE_HAVE_IOSTREAM
   #include <iostream>
   #define USE_STL_STREAM
-#elif defined (HAVE_IOSTREAM_H)
+#elif defined (OCE_HAVE_IOSTREAM_H)
   #include <iostream.h>
 #else
-  #error "check config.h file or compilation options: either HAVE_IOSTREAM or HAVE_IOSTREAM_H should be defined"
+  #error "check config.h file or compilation options: either OCE_HAVE_IOSTREAM or OCE_HAVE_IOSTREAM_H should be defined"
 #endif
 
-#ifdef HAVE_IOMANIP
+#ifdef OCE_HAVE_IOMANIP
   #include <iomanip>
-#elif defined (HAVE_IOMANIP_H)
+#elif defined (OCE_HAVE_IOMANIP_H)
   #include <iomanip.h>
 #endif
 
-#ifdef HAVE_FSTREAM
+#ifdef OCE_HAVE_FSTREAM
   #include <fstream>
-#elif defined (HAVE_FSTREAM_H)
+#elif defined (OCE_HAVE_FSTREAM_H)
   #include <fstream.h>
 #endif
 
Index: oce/inc/Standard_values.h
===================================================================
--- oce.orig/inc/Standard_values.h
+++ oce/inc/Standard_values.h
@@ -27,12 +27,14 @@
 # include <oce-config.h>
 #endif
 
-#ifdef HAVE_LIMITS
+#ifdef OCE_HAVE_CLIMITS
+# include <climits>
+#elif defined (OCE_HAVE_LIMITS)
 # include <limits>
-#elif defined (HAVE_LIMITS_H)
+#elif defined (OCE_HAVE_LIMITS_H)
 # include <limits.h>
 #else
-#error "check config.h file or compilation options: either HAVE_LIMITS or HAVE_LIMITS_H should be defined"
+#error "check config.h file or compilation options: OCE_HAVE_CLIMITS, OCE_HAVE_LIMITS, or OCE_HAVE_LIMITS_H should be defined"
 #endif
 #endif
 
Index: oce/src/BinLDrivers/BinLDrivers_DocumentRetrievalDriver.cxx
===================================================================
--- oce.orig/src/BinLDrivers/BinLDrivers_DocumentRetrievalDriver.cxx
+++ oce/src/BinLDrivers/BinLDrivers_DocumentRetrievalDriver.cxx
@@ -481,7 +481,7 @@
 						    Standard_IStream& IS)
 {
   if(!IS.eof()) {
-#if defined(WNT) || defined(HAVE_IOSTREAM)
+#if defined(WNT) || defined(OCE_HAVE_IOSTREAM)
     const Storage_Position endPos = IS.rdbuf()->pubseekoff(0L, std::ios_base::end, std::ios_base::in);
 #else
     const Storage_Position endPos = IS.rdbuf()->seekoff(0L, unsafe_ios::end, unsafe_ios::in);
Index: oce/src/Draw/Draw_BasicCommands.cxx
===================================================================
--- oce.orig/src/Draw/Draw_BasicCommands.cxx
+++ oce/src/Draw/Draw_BasicCommands.cxx
@@ -39,9 +39,9 @@
 #include <process.h>
 #include <stdio.h>
 #include <stdlib.h>
-#ifdef HAVE_LIMITS 
+#ifdef OCE_HAVE_LIMITS
 # include <limits>
-#elif defined (HAVE_LIMITS_H)
+#elif defined (OCE_HAVE_LIMITS_H)
 # include <limits.h>
 #endif
 
Index: oce/src/Graphic3d/Graphic3d_Structure.pxx
===================================================================
--- oce.orig/src/Graphic3d/Graphic3d_Structure.pxx
+++ oce/src/Graphic3d/Graphic3d_Structure.pxx
@@ -6,9 +6,9 @@
 # include <oce-config.h>
 #endif
 
-#ifdef HAVE_LIMITS
+#ifdef OCE_HAVE_LIMITS
 # include <limits>
-#elif defined (HAVE_LIMITS_H)
+#elif defined (OCE_HAVE_LIMITS_H)
 # include <limits.h>
 #endif
 
Index: oce/src/OSD/OSD_Chronometer.cxx
===================================================================
--- oce.orig/src/OSD/OSD_Chronometer.cxx
+++ oce/src/OSD/OSD_Chronometer.cxx
@@ -48,9 +48,9 @@
 #   define CLK_TCK	CLOCKS_PER_SEC
 #  endif
 
-#ifdef HAVE_LIMITS
+#ifdef OCE_HAVE_LIMITS
 # include <limits>
-#elif defined (HAVE_LIMITS_H)
+#elif defined (OCE_HAVE_LIMITS_H)
 # include <limits.h>
 #endif
 
